metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -615,3 +619,16 @@ final class FileDecoderProvider(
         )
     }
 }
+
+object URIEncoder {
```

Comment:
Finally decided to do our own encoder for things not accepted by URI.create. I can't find anything in the standard library that does it

Reply:
Now there is unnecessary code in `CommandHTMLFormat`

Reply:
Or maybe not, encoding logic is slightly different :/

Reply:
Ok, I could merge them both, but it would make the logic a bit more complicated, so I decided against, since we don't need a decoder for CommandHTMLFormat